coderd/experiments_test.go

```diff
@@ -0,0 +1,84 @@
+package coderd_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/coder/coder/coderd/coderdtest"
+	"github.com/coder/coder/codersdk"
+	"github.com/coder/coder/testutil"
+)
+
+// nolint:paralleltest  // t.Setenv
+func Test_Experiments(t *testing.T) {
+	// In these tests we set the environment variable directly, rather than
+	// modifying the deployment config. This corresponds more directly to what
+	// an end user would do. Also, modifying the deployment config doesn't work
+	// for expanding the wildcard because of how viper binds stuff.
+	t.Run("empty", func(t *testing.T) {
+		client := coderdtest.New(t, nil)
+
+		ctx, cancel := context.WithTimeout(context.Background(), testutil.WaitLong)
+		defer cancel()
+
+		experiments, err := client.Experiments(ctx)
+		require.NoError(t, err)
+		require.NotNil(t, experiments)
+		require.Empty(t, experiments)
+		require.False(t, experiments.Enabled(codersdk.ExperimentVSCodeLocal))
+		require.False(t, experiments.Enabled("foo"))
+	})
+
+	t.Run("multiple features", func(t *testing.T) {
```

Comment:
Do you think that we should consider also feature validation? We could catch issues like typo in the feature name.

Reply:
At most, I'd consider emitting a warning. Experiments may be removed at any time without notice.

Reply:
SGTM. Warning users about unsupported experiments is an option.